llmstack/server/settings.py

```diff
@@ -576,6 +576,54 @@
     },
 }
 
+ENABLE_SAML_SOCIALACCOUNT = os.getenv("ENABLE_SAML_SOCIALACCOUNT", "False") == "True"
+if ENABLE_SAML_SOCIALACCOUNT:
+    SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+
+    if "allauth.socialaccount.providers.saml" not in INSTALLED_APPS:
+        INSTALLED_APPS.append("allauth.socialaccount.providers.saml")
+
+    # Microsoft Entra ID SAML Configuration
+    SAML_APP_NAME = os.getenv("SAML_APP_NAME", "")
+    SAML_APP_CLIENT_ID = os.getenv("SAML_APP_CLIENT_ID", "")
+    SAML_APP_TENANT_ID = os.getenv("SAML_APP_TENANT_ID", "")
+    SAML_APP_CERTIFICATE = os.getenv("SAML_APP_CERTIFICATE", "")
+
+    SAML_APP = {
+        "name": SAML_APP_NAME,
+        "provider_id": f"https://login.microsoftonline.com/{SAML_APP_TENANT_ID}/",
+        "client_id": SAML_APP_CLIENT_ID,
+        "settings": {
```

Comment:
Do you want to convert these common settings into a template considering these are specific to O365 and others will have different idp etc.?

Reply:
Makes sense. Let me work on this.

Reply:
@ajhai Moved these settings into `SAML_APP_CONFIG` env variable. Please check line #601. Let me know if this is looking good.